presto-tpch/src/main/java/com/facebook/presto/tpch/statistics/StatisticsEstimator.java

```diff
@@ -67,27 +67,24 @@ else if (columnValuesRestrictions.values().stream().allMatch(List::isEmpty)) {
 
     private TableStatisticsData addPartitionStats(TableStatisticsData left, TableStatisticsData right, TpchColumn<?> partitionColumn)
     {
-        return new TableStatisticsData(
-                left.getRowCount() + right.getRowCount(),
-                addPartitionStats(left.getColumns(), right.getColumns(), partitionColumn));
-    }
-
-    private Map<String, ColumnStatisticsData> addPartitionStats(Map<String, ColumnStatisticsData> leftColumns, Map<String, ColumnStatisticsData> rightColumns, TpchColumn<?> partitionColumn)
-    {
-        return leftColumns.entrySet().stream().collect(toImmutableMap(
+        Map<String, ColumnStatisticsData> combinedColumnStats = left.getColumns().entrySet().stream().collect(toImmutableMap(
```

Comment:
I would rather tidy up this class, since you are changing it anyway. The way it is implemented right now is hard to read. It is very confusing why the `TpchColumn<?> partitionColumn, String columnName` are being passed all around. So i would either drop this commit at all, and pretend as i never see this class. Or i will try to make it tidy =)

Reply:
![refactor](https://user-images.githubusercontent.com/5570988/43268027-9014891c-90bd-11e8-9eee-79286fb07032.png)